tgui/packages/tgui/interfaces/NtosPay.tsx

```diff
@@ -0,0 +1,151 @@
+};
+let name_to_token;
+
+export const NtosPay = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const {
+    name,
+    owner_token,
+    money,
+    token,
+    money_to_send,
+    trans_list = [],
+    wanted_token,
+  } = data;
+
+  if (name) {
+    return (
+      <NtosWindow width={495} height={655}>
+        <NtosWindow.Content>
+          <Stack vertical>
+            <Section textAlign="center">
+              <Table>
+                <Table.Row>Hi, {name}.</Table.Row>
+                <Table.Row>Your pay token is {owner_token}.</Table.Row>
+                <Table.Row>
+                  Account balance: {money} credit{money === 1 ? '' : 's'}
+                </Table.Row>
+              </Table>
+            </Section>
+          </Stack>
+          <Divider hidden />
+
+          <Stack>
+            <Stack.Item>
+              <Section vertical title="Transfer Money">
+                <Box>
+                  <Tooltip
+                    content="Enter the pay token of the account you want to trasfer credits."
+                    position="top">
+                    <Input
+                      placeholder="Pay Token"
+                      width="190px"
+                      onChange={(e, value) =>
+                        act('changeValue', {
+                          token: value,
+                        })
+                      }
+                    />
+                  </Tooltip>
+                </Box>
+                <NumberInput
+                  animate
+                  unit="cr"
+                  value={money_to_send}
+                  minValue={1}
+                  maxValue={money}
+                  width="83px"
+                  onChange={(e, value) =>
+                    act('changeValue', {
+                      money_to_send: value,
+                    })
+                  }
+                />
+                <Button
+                  content="Send credits"
+                  onClick={() => act('Transaction')}
+                />
+              </Section>
+            </Stack.Item>
+            <Stack.Item>
+              <Section title="Know Token" width="270px" height="98px">
+                <Box>
+                  <Input
+                    placeholder="Full name of account."
+                    width="190px"
+                    onChange={(e, value) => (name_to_token = value)}
+                  />
+                  <Button
+                    content="Get it"
+                    onClick={() =>
+                      act('GetPayToken', {
+                        wanted_name: name_to_token,
+                      })
+                    }
+                  />
+                </Box>
+                <Divider hidden />
+                <Box nowrap>{wanted_token}</Box>
+              </Section>
+            </Stack.Item>
+          </Stack>
+
+          <Divider hidden />
+
+          <Stack vertical>
+            <Section title="Transaction History">
```

Comment:
```suggestion
          <Stack fill vertical>
            <Section scrollable title="Transaction History">
```
a bit of a personal nit pick but i think the transaction log could get bulky (if you're rich or unlucky) and should be scrollable while the primary window isn't.

** this may mean that you would need to wrap the entire return block in a stack, the subcomponents into a stack.item, and transaction history wrapped in
`<Stack.Item grow>`